administrator/components/com_templates/resources/scripts/components/app.vue

```diff
@@ -0,0 +1,378 @@
+                type: 'column',
+                options: {
+                  size: 'col-sm-' + this.column_size
+                },
+                children: [{
+                  type: 'position',
+                  options: {},
+                  children: []
+                }]
+              });
+            }
+            else {
+              alert('Grid size cannot be greter than 12');
+            }
+          }
+          this.back();
+        }
+        else {
+          this.showSettings = true;
+          this.add_column = true;
+          this.grid_selected = grid;
+        }
+      },
+      deleteColumn(grid, column) {
+        const index = grid.children.indexOf(column);
+        if (index > -1) {
+          grid.children.splice(index, 1);
+        }
+      },
+      editPosition(grid, column, submit) {
+        if (submit) {
+          if (this.module_chrome !== 'none') {
+            this.column_selected.children[0].options.module_chrome = this.module_chrome;
+          }
+          this.module_chrome = 'none';
+          this.back();
+        }
+        else {
+          this.showSettings = true;
+          this.edit_position = true;
+          this.column_selected = column;
+          this.grid_selected = grid;
+        }
+      },
+      editColumn(grid, column, submit) {
+        if (submit) {
+          if (this.column_class !== '') {
+            const index = this.grid_selected.children.indexOf(this.column_selected);
+            if (index > -1) {
+              this.grid_selected.children[index].options.class = this.column_class;
+            }
+          }
+          this.back();
+        }
+        else {
+          this.edit_column = true;
+          this.showSettings = true;
+          this.column_selected = column;
+          this.grid_selected = grid;
+        }
+      },
+      editGrid(grid, submit) {
+        if (submit) {
+          if (this.grid_class !== '') {
+            this.grid_selected.options.class = this.grid_class;
+          }
+
+          if (this.column_size !== '') {
+            let sum = 0;
+            this.grid_selected.children.forEach(element => {
+              sum += Number(element.options.size.split('-')[2]);
+            });
+            if (sum + Number(this.column_size) <= 12) {
+              this.grid_selected.children.push({
+                type: 'column',
+                options: {
+                  size: 'col-sm-' + this.column_size
+                },
+                children: [{
+                  type: 'position',
+                  options: {},
+                  children: []
+                }]
+              });
+            }
+            else {
+              alert('Grid size cannot be greater than 12');
+            }
+          }
+          this.back();
+        }
+        else {
+          this.edit_grid = true;
+          this.grid_selected = grid;
+          this.showSettings = true;
+        }
+      },
+      back() {
```

Comment:
If `back()` function is something like reset then i would suggest you to change it to `reset()`